internal/cobraprompt/prompt.go

```diff
@@ -146,18 +151,27 @@ func (co CobraPrompt) Run(ctx context.Context, root *cobra.Command, cnfg *hazelc
 			root, _ = rootcmd.New(cnfg)
 			prepareRootCmdForPrompt(co, root)
 			root.SetArgs(promptArgs)
+			root.SetFlagErrorFunc(func(_ *cobra.Command, err error) error {
+				return hzcerror.FlagError(err)
+			})
 			if err := root.ExecuteContext(ctx); err != nil {
 				if errors.Is(err, ErrExit) {
 					exitPromptSafely()
 					return
 				}
+				if err.Error() == `required flag(s) "name" not set` {
```

Comment:
Isn't there a better way than matching to the error string?

Reply:
Sadly no, cobra creates this error with `fmt.Errorf`

Reply:
Could you add a TODO to this line to try to find a better solution? I think the code in this line is very fragile and could produce bugs.

Reply:
Makes ton of sense! TODO added https://github.com/hazelcast/hazelcast-commandline-client/pull/61/commits/4cff1e6baac2c26537a010cf79a33bc6cf07c476